proto/error/v1/error.proto

```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+package nitric.error.v1;
+
+// protoc plugin options for code generation
+option go_package = "nitric/v1;v1";
+option java_package = "io.nitric.proto.error.v1";
+option java_multiple_files = true;
+option java_outer_classname = "Errors";
+option php_namespace = "Nitric\\Proto\\Error\\V1";
+option csharp_namespace = "Nitric.Proto.Error.v1";
+
+message ErrorScope {
+    // The API service invoked, e.g. 'Service.Method'.
+    string service = 1;
+
+    // The plugin method invoked, e.g. 'PluginService.Method'.
+    string plugin = 2;
+
+    // The plugin method arguments, ensure only non-sensitive data is specified.
+    map<string, string> args = 3;
+}
+
+message ErrorDetails {
+    // The developer error message, explaining the error and ideally solution.
+    string message = 1;
+
+    // The cause of the error.
+    string cause = 2;
```

Comment:
Just a thought on this, would we want cause to be chainable? If unsure we might be able to get away with leaving this out in the first iteration, and adding in cause later (it's easier to leave things out and add them later than it is to refactor with backwards compatibility.

Reply:
What did you have in mind by chainable, creating a linked list structure? I would probably recommend we walk the cause chain in the membrane grpc/errors and have it assign the root cause, if available to the cause field.

I have found the cause to be really useful as it often contains the underlying plugin service error.